cuda_bindings/cuda/bindings/_path_finder/load_dl_windows.py

```diff
@@ -0,0 +1,141 @@
+WINBASE_LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR = 0x00000100
+WINBASE_LOAD_LIBRARY_SEARCH_DEFAULT_DIRS = 0x00001000
+
+
+def abs_path_for_dynamic_library(handle: int) -> str:
+    """Get the absolute path of a loaded dynamic library on Windows.
+
+    Args:
+        handle: The library handle
+
+    Returns:
+        The absolute path to the DLL file
+
+    Raises:
+        OSError: If GetModuleFileNameW fails
+    """
+    buf = ctypes.create_unicode_buffer(260)
+    n_chars = ctypes.windll.kernel32.GetModuleFileNameW(ctypes.wintypes.HMODULE(handle), buf, len(buf))
+    if n_chars == 0:
+        raise OSError("GetModuleFileNameW failed")
+    return buf.value
+
+
+@functools.cache
+def cuDriverGetVersion() -> int:
+    """Get the CUDA driver version.
+
+    Returns:
+        The CUDA driver version number
+
+    Raises:
+        AssertionError: If the driver version cannot be obtained
+    """
+    handle = win32api.LoadLibrary("nvcuda.dll")
+
+    kernel32 = ctypes.WinDLL("kernel32", use_last_error=True)
+    GetProcAddress = kernel32.GetProcAddress
+    GetProcAddress.argtypes = [ctypes.wintypes.HMODULE, ctypes.wintypes.LPCSTR]
+    GetProcAddress.restype = ctypes.c_void_p
+    cuDriverGetVersion = GetProcAddress(handle, b"cuDriverGetVersion")
+    assert cuDriverGetVersion
+
+    FUNC_TYPE = ctypes.CFUNCTYPE(ctypes.c_int, ctypes.POINTER(ctypes.c_int))
+    cuDriverGetVersion_fn = FUNC_TYPE(cuDriverGetVersion)
+    driver_ver = ctypes.c_int()
+    err = cuDriverGetVersion_fn(ctypes.byref(driver_ver))
+    assert err == 0
+    return driver_ver.value
+
+
+def check_if_already_loaded(libname: str) -> Optional[LoadedDL]:
+    """Check if the library is already loaded in the process.
+
+    Args:
+        libname: The name of the library to check
+
+    Returns:
+        A LoadedDL object if the library is already loaded, None otherwise
+
+    Example:
+        >>> loaded = check_if_already_loaded("cudart")
+        >>> if loaded is not None:
+        ...     print(f"Library already loaded from {loaded.abs_path}")
+    """
+    from .supported_libs import SUPPORTED_WINDOWS_DLLS
+
+    for dll_name in SUPPORTED_WINDOWS_DLLS.get(libname, ()):
+        try:
+            handle = win32api.GetModuleHandle(dll_name)
+        except pywintypes.error:
+            continue
+        else:
+            return LoadedDL(handle, abs_path_for_dynamic_library(handle), True)
+    return None
+
+
+def load_with_system_search(name: str, _unused: str) -> Optional[LoadedDL]:
+    """Try to load a DLL using system search paths.
+
+    Args:
+        name: The name of the library to load
+        _unused: Unused parameter (kept for interface consistency)
+
+    Returns:
+        A LoadedDL object if successful, None if the library cannot be loaded
+    """
+    from .supported_libs import SUPPORTED_WINDOWS_DLLS
+
+    driver_ver = cuDriverGetVersion()
+    del driver_ver  # Keeping this here because it will probably be needed in the future.
```

Comment:
Why are we doing this?

Reply:
I removed the `cuDriverGetVersion()` function for now: commit 4ce94be563f5dba48ead76eeb992bb7c041890ca

It's easy enough to put it back later as needed.